code/game/machinery/doors/firedoor.dm

```diff
@@ -650,19 +674,22 @@
 
 /obj/structure/firelock_frame/heavy
 	name = "heavy firelock frame"
-	reinforced = TRUE
+	firelock_type = /obj/machinery/door/firedoor/heavy
 
 /obj/structure/firelock_frame/border
 	name = "firelock frame"
 	icon = 'icons/obj/doors/edge_Doorfire.dmi'
-	icon_state = "frame1"
-	firelock_type = /obj/machinery/door/firedoor/border_only/closed
+	icon_state = "door_frame"
+	firelock_type = /obj/machinery/door/firedoor/border_only
 	flags_1 = ON_BORDER_1
 
 /obj/structure/firelock_frame/border/ComponentInitialize()
 	. = ..()
 	AddComponent(/datum/component/simple_rotation, ROTATION_ALTCLICK | ROTATION_CLOCKWISE | ROTATION_COUNTERCLOCKWISE | ROTATION_VERBS, null, CALLBACK(src, .proc/can_be_rotated))
 
+/obj/structure/firelock_frame/border/update_icon()
```

Comment:
update_icon should not be overriden in this manner. Look at the base definition of it to see why.